### PR TITLE
GLTF - Load Skin Async

### PR DESCRIPTION
**Pull request by MackeyK24**
GLTF2 - Custom Load Skin Async Support

Yo @bghgary ... Please take a look at :)

### CODE REVIEW

File: loaders/src/glTF/2.0/glTFLoaderExtension.ts [hunk 1]
```diff
@@ -104,6 +104,15 @@ export interface IGLTFLoaderExtension extends IGLTFBaseLoaderExtension, IDisposa
...
     loadAnimationAsync?(context: string, animation: IAnimation): Nullable<Promise<AnimationGroup>>;
 
+    /**
+     * Define this method to modify the default behavior when loading skins.
+     * @param context The context when loading the asset
+     * @param node The glTF node property
+     * @param skin The glTF skin property
+     * @returns A promise that resolves with the loaded Babylon skeleton when the load is complete or null if not handled
+     */
+    loadSkinAsync?(context: string, node: INode, skin: ISkin): Nullable<Promise<void>>;
```
Comment (bghgary) [inline loaders/src/glTF/2.0/glTFLoaderExtension.ts]: Let's make this an underscore function for now. The questions for this contract is whether we need an assign function like the some of the other functions and whether the skeleton should be returned in the promise. I think the answer is yes to both.

```suggestion
    _loadSkinAsync?(context: string, node: INode, skin: ISkin): Nullable<Promise<void>>;
```

Comment (MackeyK24) [reply loaders/src/glTF/2.0/glTFLoaderExtension.ts]: Changes Made :)

File: dist/preview release/what's new.md [hunk 2]
```diff
@@ -134,6 +134,7 @@
 - Added LoadScriptAsync tools helper function [MackeyK24](https://github.com/mackeyk24))  
 - Added customShaderNameResolve to PBRMaterialBase to allow subclasses to specify custom shader information [MackeyK24](https://github.com/mackeyk24))
 - Added PBRCustomMaterial to material library to allow easy subclassing of PBR materials [MackeyK24](https://github.com/mackeyk24))
+- Added LoadSkinAsync to GLTF2 loaders to allow custom skinning support [MackeyK24](https://github.com/mackeyk24))
```
Comment (bghgary) [inline dist/preview release/what's new.md]: Should probably remove this until the underscore is removed (i.e. the contract is solid)
```suggestion
```

File: loaders/src/glTF/2.0/glTFLoaderExtension.ts [hunk 3]
```diff
@@ -104,6 +104,15 @@ export interface IGLTFLoaderExtension extends IGLTFBaseLoaderExtension, IDisposa
      */
     loadAnimationAsync?(context: string, animation: IAnimation): Nullable<Promise<AnimationGroup>>;
 
+    /**
+     * Define this method to modify the default behavior when loading skins.
+     * @param context The context when loading the asset
+     * @param node The glTF node property
+     * @param skin The glTF skin property
+     * @returns A promise that resolves with the loaded Babylon skeleton when the load is complete or null if not handled
```
Comment (bghgary) [inline loaders/src/glTF/2.0/glTFLoaderExtension.ts]:
```suggestion
     * @returns A promise that resolves when the load is complete or null if not handled
```

**Comment by bghgary** [comment]
@MackeyK24 Two minor comments. LGTM otherwise.

**Comment by MackeyK24** [comment]
Dunno about the build errors:

1... The change i made is a work in progress _loadSkinAsync function that is not ready for WHATS NEW

2... Says unnecessary semicolon in transformNode.ts... I never made changes to transformNode.ts so i don't know what thats all about

But @bghgary  approved this PR :)

**Comment by deltakosh** [comment]
What's new is not a big deal but linting is. Please run "npm build run" to fix that (and do not forget to revert all changes that will occur in /dist)

**Comment by deltakosh** [comment]
Nevermind I'll take care of the linting

**Pull request merged**